flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -416,7 +416,9 @@ private void applyDescriptorOptionToConfig(final CommandLine commandLine, final
 		}
 
 		final String dynamicPropertiesEncoded = encodeDynamicProperties(commandLine);
-		configuration.setString(YarnConfigOptionsInternal.DYNAMIC_PROPERTIES, dynamicPropertiesEncoded);
+		if (dynamicPropertiesEncoded != null && !dynamicPropertiesEncoded.isEmpty()) {
```

Comment:
Is it a relevant diff(i.e., cause tests fail) or just improving on the fly? I'm ok with on the fly improvement but to be clear.

Reply:
This is not related to this PR, I agree, this is why it is marked as a hotifx.